src/ios/CDVDevice.m

```diff
@@ -86,10 +86,14 @@ - (void)getDeviceInfo:(CDVInvokedUrlCommand*)command
 - (NSDictionary*)deviceProperties
 {
     UIDevice* device = [UIDevice currentDevice];
+	//strip non-alphanumeric from device name
+    NSCharacterSet *charactersToRemove = [[NSCharacterSet alphanumericCharacterSet] invertedSet];
+    NSString *deviceName = [[[device name] componentsSeparatedByCharactersInSet:charactersToRemove] componentsJoinedByString:@""];
```

Comment:
Apple's documentation for [UIDevice->name](https://developer.apple.com/documentation/uikit/uidevice/1620015-name?language=objc) states

> The value of this property is an arbitrary alphanumeric string that is associated with the device as an identifier. For example, you can find the name of an iOS device in the General > About settings.

So is scanning for non-alphanumeric characters really necessarily? Apple specifically states it's an alphanumeric string.

Reply:
True, Apple's documentation states "alphanumeric string" but from experience it is not so.
You can go change your device's name to anything you want (ie no forbidden characters).

Reply:
I see, thanks for the clarification